### PR TITLE
feat: add `owner_oidc_access_token` to `coder_workspace` data source

### PR DESCRIPTION
**Pull request by kylecarbs**
See the discussion in Discord here:
https://discord.com/channels/747933592273027093/1071182088490987542/1071182088490987542

Related provider PR: coder/terraform-provider-coder#91

### CODE REVIEW

File: provisionersdk/proto/provisioner.proto
Comment (deansheather) [inline provisionersdk/proto/provisioner.proto:204]: tabs vs spaces

**Review by deansheather — COMMENTED** [review]
This needs to be configurable via a flag. I think it should be disabled by default personally, but at the very least it needs to be the other way around IMO.

**Comment by kylecarbs** [comment]
@deansheather why does this need to be configurable via a flag?

**Comment by deansheather** [comment]
To prevent template admins from reading OIDC secrets without deployment approval. It seems dangerous to allow anyone with write access to templates to push an update that starts reading OIDC tokens and doing nefarious things with them.

We have a similar feature in v1 and it's disabled by default for this reason

**Comment by kylecarbs** [comment]
Hmm, fair. I suppose this would permit any template author to gain access to users inside of Coder, so it could be real bad. Some additional thought is needed on how we do it... :thinking:

**Comment by bpmct** [comment]
Can we just make it disabled by default?

**Comment by bpmct** [comment]
We'll need to rethink our template authorship experience since the `Template Admin` role is already quite risky, [as we have documented here](https://coder.com/docs/v2/latest/admin/users#security-notes).

However, that shouldn't block this enhancement. Let's move forward by making this configurable by a deployment-wide flag that is disabled by default.

**Review by bpmct — APPROVED** [review]

**Pull request merged**
